core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/visualization/barChart/BarChartOpDesc.scala

```diff
@@ -0,0 +1,170 @@
+package edu.uci.ics.texera.workflow.operators.visualization.barChart
+
+import com.fasterxml.jackson.annotation.{JsonIgnore, JsonProperty, JsonPropertyDescription}
+import edu.uci.ics.amber.engine.operators.OpExecConfig
+import edu.uci.ics.texera.workflow.common.metadata.{
+  InputPort,
+  OperatorGroupConstants,
+  OperatorInfo,
+  OutputPort
+}
+import edu.uci.ics.texera.workflow.common.metadata.annotations.{
+  AutofillAttributeName,
+  AutofillAttributeNameList
+}
+import edu.uci.ics.texera.workflow.common.operators.aggregate.DistributedAggregation
+import edu.uci.ics.texera.workflow.common.tuple.Tuple
+import edu.uci.ics.texera.workflow.common.tuple.schema.AttributeTypeUtils.parseTimestamp
+import edu.uci.ics.texera.workflow.common.tuple.schema.{
+  Attribute,
+  AttributeType,
+  OperatorSchemaInfo,
+  Schema
+}
+import edu.uci.ics.texera.workflow.operators.visualization.{
+  AggregatedVizOpExecConfig,
+  VisualizationConstants,
+  VisualizationOperator
+}
+
+import java.util.Collections.singletonList
+import scala.jdk.CollectionConverters.asScalaBuffer
+
+/**
+  * Supports a bar chart with internal aggregation for one name column (label, x-axis) and multiple data columns (y-axis).
+  * If no data column is provided, the count of each label is returned; else the aggregated sum over each data column,
+  * grouped by each label is returned.
+  */
+class BarChartOpDesc extends VisualizationOperator {
+  @JsonProperty(value = "name column", required = true)
+  @JsonPropertyDescription("column of name (for x-axis)")
+  @AutofillAttributeName var nameColumn: String = _
+
+  @JsonProperty(value = "data column(s)", required = false)
+  @JsonPropertyDescription("column(s) of data (for y-axis)")
+  @AutofillAttributeNameList var dataColumns: List[String] = _
+
+  @JsonIgnore
+  private var groupBySchema: Schema = _
+  @JsonIgnore
+  private var finalAggValueSchema: Schema = _
+
+  override def chartType: String = VisualizationConstants.BAR
+
+  def noDataCol: Boolean = dataColumns == null || dataColumns.isEmpty
+
+  def resultAttributeNames: List[String] = if (noDataCol) List("count") else dataColumns
+
+  override def operatorExecutor(operatorSchemaInfo: OperatorSchemaInfo): OpExecConfig = {
+    if (nameColumn == null || nameColumn == "") {
+      throw new RuntimeException("bar chart: name column is null or empty")
+    }
+
+    this.groupBySchema = getGroupByKeysSchema(operatorSchemaInfo.inputSchemas)
+    this.finalAggValueSchema = getFinalAggValueSchema
+
+    val aggregation =
+      if (noDataCol)
+        new DistributedAggregation[Integer](
```

Comment:
Do we want to make some utility functions such as `DistributedCount[T](columns)`, `DistributedSum[T](columns)` in future PRs?

Reply:
Yes, but not in this PR.